hail/python/hail/backend/local_backend.py

```diff
@@ -183,27 +180,31 @@ def hail_package(self):
     def utils_package_object(self):
         return self._utils_package_object
 
+    def register_ir_function(self,
```

Comment:
move to py4jbackend? looks shared with spark

Reply:
Not sure if this is intentional, but this differs slightly from the spark backend. It was previously in py4j_backend and overridden in the spark_backend so I instead just moved this to local_backend. The difference is that spark_backend asserts the ir does not use randomness and does not call `finalize_randomness`. Happy to change if these should not in fact be different.

Reply:
@patrick-schultz can you comment here in case you know so I don't have to dig in?

Reply:
I think an easy fix here is to handle randomness in `define_function` by adding an extra parameter `__rng_state` (if the body uses randomness, but it doesn't hurt to do it all the time), then making the returned `f` pass in `Ref(__rng_state)` for the parameter, to capture the state from the caller's environment. Then `register_ir_function` should not call `finalize_randomness`.